src/main/scala/intellij/haskell/runconfig/test/HaskellTestConfiguration.scala

```diff
@@ -45,8 +45,8 @@ class HaskellTestConfiguration(override val name: String, override val project:
   //https://github.com/commercialhaskell/stack/issues/2210
   override def getState(executor: Executor, environment: ExecutionEnvironment): HaskellStackStateBase = {
     val packageName: String = CabalQuery.getPackageName(project).getOrElse(project.getName)
-    val command = s"test $packageName:$myTestsuite" ++
-      (if (getTestFilter.isEmpty) "" else s" --test-arguments '-m${'"'}$getTestFilter${'"'}'")
-    new HaskellStackStateBase(this, environment, command)
+    val parameters = List("test", s"$packageName:$myTestsuite") ++
```

Comment:
Ah, the test filter is quite straightforward, it can fuzzy filter test cases by its description string, you can see the screenshot below, it is quite useful for production code base when it contains thousands of test cases, you won't want to run them all every time 😄 

![screenshot](https://cloud.githubusercontent.com/assets/6234553/23558101/598e17c8-006d-11e7-913d-b9b5ae9022d3.gif)